src/pdn/src/PdnGen.tcl

```diff
@@ -399,6 +399,47 @@ variable voltage_domains {
     primary_power VDD primary_ground VSS
   }
 }
+# It is better to get min and max density from odb.
+# But I cannot find any API.
+variable min_density 66.6
+variable max_density 75.0
+variable default_maxwidth 10
+
+proc get_max_width {layer} {
+  set max_width [$layer getMaxWidth]
+  if {$max_width < 2147483647} {
+    # OpenDB hides the fact that the layer has no MAXWIDTH field 
+    # and returns the biggest value of int32_t.
+    # So I have to guess if it is set.
+    return $max_width
+  }
+
+  # For some unknown reason, it is possible for a layer to has no MAXWIDTH field.
+  # In this case, we will try MAXWIDTH in other layers.
+  # If there is no MAXWIDTH in any layer, a default value will be used.
+  variable tech
+  set layers [$tech getLayers]
+  foreach layer $layers {
+    set max_width [$layer getMaxWidth]
+    if {$max_width < 2147483647} {
+      return $max_width
+    }
+  }
+
+  variable default_maxwidth
+  return $default_maxwidth
+}
```

Comment:
If there is no MAXWIDTH rule defined for the layer, then the width of the pin should be used.